include/oneapi/dpl/pstl/hetero/numeric_impl_hetero.h

```diff
@@ -155,6 +155,28 @@ __pattern_transform_scan_base_impl(_ExecutionPolicy&& __exec, _Iterator1 __first
         .wait();
 }
 
+template <typename Iterator>
+bool
+__is_equal_iterators_impl(Iterator it1, Iterator it2)
+{
+    return it1 == it2;
+}
+
+template <typename Iterator1, typename Iterator2>
+bool
+__is_equal_iterators_impl(Iterator1 it1, Iterator2 it2)
+{
+    static_assert("Unable to compare the iterators of different types");
```

Comment:
Should be another text, like this:
"In-place exclusive scan works correctly only if an input and an output iterators are the same type."

Reply:
Fixed